### PR TITLE
Some updates

### PR DESCRIPTION
**Pull request by jcapriot**
This cleans out all of the notebooks' cells, as well as updated a few non-running notebooks.

There's also a lot of cleanup in the dc simulations to take advantage of the improved fields object, and make use of caching the simulations when possible to make viewing plots a little more responsive.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -19,9 +19,19 @@ env:
...
+before_install:
+# Install packages
+  - wget https://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh -O miniconda.sh;
+  - chmod +x miniconda.sh
+  - ./miniconda.sh -b -p $HOME/miniconda
+  - export PATH=/home/travis/anaconda/bin:/home/travis/miniconda/bin:$PATH
+  - conda update --yes conda
+
 install:
+  - conda env create -f environment-dev.yml
```
Comment (lheagy) [inline .travis.yml:32]: Thanks for this Joe!

File: geoscilabs/dcip/DCWidgetPlate2_5D.py [hunk 2]
```diff
@@ -53,51 +55,77 @@
     & (mesh.gridFy[:, 1] <= ymax)
 )
 indF = np.concatenate((indx, indy))
+_cache = {
```
Comment (lheagy) [inline geoscilabs/dcip/DCWidgetPlate2_5D.py:58]: good call on these!

File: geoscilabs/dcip/DCWidgetPlate2_5D.py [hunk 3]
```diff
@@ -53,51 +55,77 @@
...
+        or _cache["xc"] != xc
+        or _cache["zc"] != zc
+        or _cache["rotAng"] != rotAng
+        or _cache["sigplate"] != sigplate
+        or _cache["sighalf"] != sighalf
+    )
+    if re_run:
+        # Create halfspace model
+        mhalf = np.log(sighalf * np.ones([mesh.nC]))
+        # mhalf = sighalf*np.ones([mesh.nC,])
```
Comment (lheagy) [inline geoscilabs/dcip/DCWidgetPlate2_5D.py:86]: do you think we still need these (and similarly on 94, etc)? If not, we could clear them out

File: geoscilabs/dcip/DCWidgetResLayer2D.py [hunk 4]
```diff
@@ -646,12 +689,12 @@ def plot_Surface_Potentials(
...
         u[ind] + eps,
-        vType=xtype,
+        v_type=xtype,
         ax=ax[1],
         grid=False,
         view=view,
-        streamOpts=streamOpts,
-        pcolorOpts=pcolorOpts,
+        stream_opts=streamOpts,
+        pcolor_opts=pcolorOpts,
```
Comment (lheagy) [inline geoscilabs/dcip/DCWidgetResLayer2D.py:697]: Thank you 😄

File: geoscilabs/em/TDEMGroundedSource.py [hunk 5]
```diff
@@ -85,9 +85,9 @@ def choose_model(model):
 def run_simulation(fname="tdem_gs_half.h5", sigma_block=0.01, sigma_halfspace=0.01):
     from SimPEG.electromagnetics import time_domain as tdem
     from SimPEG.electromagnetics.utils import waveform_utils
-    from SimPEG.EM import Analytics, mu_0
+    from SimPEG.electromagnetics import mu_0
```
Comment (lheagy) [inline geoscilabs/em/TDEMGroundedSource.py:88]: we should probably just import directly from scipy here 
```suggestion
    from scipy.constants import mu_0
```

**Comment by lheagy** [comment]
It looks like there is one other file that got added in the mag directory along the way that I think could be removed? 

image

**Comment by jcapriot** [comment]
Thanks Lindsey, done and done

**Comment by lheagy** [comment]
Many thanks @jcapriot!

**Pull request merged**